src/Pages/ProductList/ProductList.component.js

```diff
@@ -3,15 +3,87 @@ import React, { Component } from "react";
 import "./ProductList.styles.scss";
 import ProductCard from "./ProductCard/ProductCard.component";
 
+const API = "http://localhost:3000/data/productlist/productlist.json";
+// const API = "http://10.58.4.20:8000/category?main=1"; //Backend API
+
 class ProductList extends Component {
+  state = {
+    isCategoryClick: false,
+    isSortingClick: false,
+    categories: "",
+    products: "",
+  };
+
+  handleSorting = () => {
+    const { isSortingClick } = this.state;
+    this.setState({
+      isSortingClick: !isSortingClick,
+    });
+  };
+
+  componentDidMount = async () => {
+    const response = await fetch(API);
+    // custom error handler
+    if (response.status !== 200) {
+      throw new Error("cannot fetch the data");
+    }
+    const data = await response.json();
+    this.setState({ categories: data.categories, products: data.products });
+  };
+
   render() {
+    const { isSortingClick, categories, products } = this.state;
     return (
       <section className="ProductList">
-        <div className="row">
-          <div className="col s12 m4">
-            <ProductCard />
+        <div className="category-nav">
+          <div className="category-title">
+            <i className="far fa-carrot"></i>
+            <span>채소</span>
+          </div>
+          <div className="category-list">
+            <ul>
+              {categories.length &&
+                categories.map((category) => (
+                  <li key={category.id}>{category.name}</li>
+                ))}
```

Comment:
```suggestion
              {categories?.map(({ id, name }) => (
                  <li key={id}>{name}</li>
                ))}
```
- optional chaining을 사용하면 보다 간편하게 falsy한 값을 처리할 수 있습니다 ([링크](https://ko.javascript.info/optional-chaining))
- 함수 파라미터에서 직접 구조분해할당이 가능합니다

Reply:
```js
  state = {
    isCategoryClick: false,
    isSortingClick: false,
    categories: [],
    products: [],
    sortings: [],
    activeSorting: 0,
  };

 {categories.length && categories.map(({ id, name }) => <li key={id}>{name}</li>)}
```
제가 이전에는 categories null값 처리하기 위해 "" 썻는데 정확히는 array 를 받아오는 거라 빈 array 를 선언해주고 true/false 판단을 위해 length는 그대로 나두고 사용해봤습니당~🤟🏻🤟🏻🤟🏻
array로 받는 항목에 대한 null값 처리를 어떻게 하는게 유리할까요? 한글자라도 짧게하려면 ""로 받아서 바로 length없이 받으면 좋을것 같긴한데....

Reply:
@VannsKang 빈 배열로 선언해놓고 map을 사용하는 경우라면 에러가 발생할 여지가 없으므로 length를 체크할 필요도 사실 애초부터 없었던 것이 아닐까 하는 생각이 들긴 하네요. 빈 스트링에 대한 내용은 잘 이해가 안되는데 좀 더 구체적으로 설명해주시면 좋을 것 같습니다.